lib/generators/declare_schema/migration/migration_generator.rb

```diff
@@ -100,7 +100,7 @@ def migrate
     if ActiveSupport::VERSION::MAJOR < 5
       def indent(string, columns)
         whitespace = ' ' * columns
-        string.gsub("\n", "\n#{whitespace}").gsub!(/ +\n/, "\n")
+        string.gsub("\n", "\n#{whitespace}").gsub(/ +\n/, "\n")
```

Comment:
Here's the bug. One `!`. At the last minute I changed to `gsub!` as a tiny optimization, since the string has already been dup'd by the previous `gsub`. Bad idea! In the event that the pattern isn't found, `gsub!` returns `nil` instead of the original string. I had no idea of that weird contract.

This would only fail in Rails 4, and only on migrations that were a single line.

Reply:
Haha glad my use case was the perfect example to spot this bug!

Reply:
Didn't understand what you meant by "weird contract" so I ran it through irb, and yeah, this is sort of infuriating. Good spot.
```
irb(main):001:0> "pizza".gsub("party", "123")
=> "pizza"
irb(main):002:0> "pizza".gsub!("party", "123")
=> nil
```

Reply:
Yeah, that was a bizarre contract choice. I checked and it is documented, though. I guess the reason I didn't know is that I almost always avoid the `!` methods.